src/EasyReadScorer.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2011-2016, Pacific Biosciences of California, Inc.
+//
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted (subject to the limitations in the
+// disclaimer below) provided that the following conditions are met:
+//
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//
+//  * Redistributions in binary form must reproduce the above
+//    copyright notice, this list of conditions and the following
+//    disclaimer in the documentation and/or other materials provided
+//    with the distribution.
+//
+//  * Neither the name of Pacific Biosciences nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// NO EXPRESS OR IMPLIED LICENSES TO ANY PARTY'S PATENT RIGHTS ARE
+// GRANTED BY THIS LICENSE. THIS SOFTWARE IS PROVIDED BY PACIFIC
+// BIOSCIENCES AND ITS CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
+// WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+// OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL PACIFIC BIOSCIENCES OR ITS
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+// USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+// ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+// OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+// SUCH DAMAGE.
+
+// Author: David Alexander
+
+#include "ModelFactory.h"
+
+#include <pacbio/consensus/EasyReadScorer.h>
+#include <pacbio/consensus/Evaluator.h>
+#include <pacbio/consensus/Template.h>
+#include <pacbio/data/Read.h>
+
+#include <memory>
+#include <string>
+
+namespace PacBio {
+namespace Consensus {
+
+class ModelConfig;
+
+Evaluator* EasyReadScorer::MakeEvaluator(
+    const std::string& tplString,
+    const PacBio::Data::MappedRead& mappedRead,
+    double minZScore,
+    double scoreDiff)
+{
+    // Make a "ModelConfig"
+    std::unique_ptr<ModelConfig> cfg = ModelFactory::Create(mappedRead.Model,
+                                                            mappedRead.SignalToNoise);
+    // make a "template" object
+    std::unique_ptr<AbstractTemplate> tpl(
+            new Template(tplString, std::move(cfg)));
+
+    return new Evaluator(std::move(tpl), mappedRead,  minZScore, scoreDiff);
```

Comment:
For your first comment: no because SWIG.

For your second comment: it doesn't really make a difference, the %newobject will cause the host language to delete it when approprate